packages/teams-js/src/public/thirdPartyCloudStorage.ts

```diff
@@ -0,0 +1,201 @@
+ * @beta
+ */
+export namespace thirdPartyCloudStorage {
+  /** Get context callback function type */
+  const files: FilesFor3PApps[] = [];
+  let helper: AttachmentListHelper = {
+    fileType: '',
+    assembleAttachment: [],
+  };
+
+  /**
+   * Object used to represent a file
+   * @beta
+   *
+   */
+  export interface FilesFor3PApps extends Blob {
+    /**
+     * A number that represents the number of milliseconds since the Unix epoch
+     */
+    lastModified: number;
+    /**
+     * Name of the file
+     */
+    name: string;
+    /**
+     * file type
+     */
+    type: string;
+    /**
+     * A string containing the path of the file relative to the ancestor directory the user selected
+     */
+    webkitRelativePath?: string;
+  }
+
+  /**
+   * File chunks an output of getDragAndDropFiles API from platform
+   */
+  export interface FileChunk {
+    /**
+     * Base 64 data for the requested uri
+     */
+    chunk: string;
+    /**
+     * chunk sequence number
+     */
+    chunkSequence: number;
+  }
+
+  /**
+   * Output of getDragAndDropFiles API from platform
+   */
+  export interface FileResult {
+    /**
+     * Error encountered in getDragAndDropFiles API
+     */
+    error?: SdkError;
+    /**
+     * File chunk which will be assemebled and converted into a blob
+     */
+    fileChunk: FileChunk;
+    /**
+     * File index of the file for which chunk data is getting recieved
+     */
+    fileIndex: number;
+    /**
+     * File type which is getting recieved
+     */
+    fileType: string;
+    /**
+     * Indicates whether this file is the last one in a sequence.
+     */
+    isLastFile: boolean;
+    /**
+     * The name of the file.
+     */
+    fileName: string;
+  }
+
+  /**
+   * Helper class for assembling files
+   */
+  export interface AttachmentListHelper {
+    /** A string representing the MIME type of the file */
+    fileType: string;
+    /** An array of {@link media.AssembleAttachment | AssembleAttachment} objects representing the media files to be sent as attachment */
+    assembleAttachment: media.AssembleAttachment[];
+  }
+
+  /**
+   * Defines the Callback function received from Third Party App
+   */
+  export interface ThirdPartyAppCallback {
+    /** Callback from third party app */
+    (files: FilesFor3PApps[], error?: SdkError): void;
+  }
+
+  let callback: ThirdPartyAppCallback | null = null;
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param {string} dragAndDropInput - The input related to the drag-and-drop operation.
+   * @param {ThirdPartyAppCallback} thirdPartycallback - callback
+   *   A callback function to handle the result of the operation
+   */
+  export function getDragAndDropFiles(dragAndDropInput: string, thirdPartycallback: ThirdPartyAppCallback): void {
+    if (!thirdPartycallback) {
+      throw new Error('[getDragAndDropFiles] Callback cannot be null');
+    }
+    callback = thirdPartycallback;
```

Comment:
'getDragAndDropFiles' API is called only once by the third-party apps and this callback we are getting from the third-party app and which we will be calling after we receive all the file from TMP.

Here I am storing callback received from third-party app in 'callback' so the I am able to get this inside 'handleGetDragAndDropFilesCallbackRequest'.

Reply:
I know your _intention_ is for people to only call it once, but there's nothing enforcing that. If someone calls it more than once it will work in an unexpected way.

Reply:
This is still unresolved. If this callback can only ever be called once successfully you this function should throw an error with helpful message if someone tries to call it a second time. That would mean whenever callback is already set when the function is called.

Reply:
Added a check if this API is called twice.